librad/src/net/peer.rs

```diff
@@ -195,9 +196,13 @@ where
         urn: RadUrn,
     ) -> impl Future<Output = impl futures::Stream<Item = PeerInfo<IpAddr>>> {
         let protocol = self.protocol.clone();
-        let urn2 = urn.clone();
+        let target_urn = urn.clone();
+        let span = tracing::trace_span!("Peer::providers");
```

Comment:
```suggestion
        let span = tracing::trace_span!("Peer::providers", urn = %urn);
```

Would also put it at the start of the method.